python/deltalake/table.py

```diff
@@ -412,35 +521,48 @@ def to_pyarrow_table(
         partitions: Optional[List[Tuple[str, str, Any]]] = None,
         columns: Optional[List[str]] = None,
         filesystem: Optional[Union[str, pa_fs.FileSystem]] = None,
+        filters: Optional[
+            Union[List[Tuple[str, str, Any]], List[List[Tuple[str, str, Any]]]]
+        ] = None,
     ) -> pyarrow.Table:
         """
         Build a PyArrow Table using data from the DeltaTable.
 
         :param partitions: A list of partition filters, see help(DeltaTable.files_by_partitions) for filter syntax
         :param columns: The columns to project. This can be a list of column names to include (order and duplicates will be preserved)
         :param filesystem: A concrete implementation of the Pyarrow FileSystem or a fsspec-compatible interface. If None, the first file path will be used to determine the right FileSystem
+        :param filters: A disjunctive normal form (DNF) for filter pushdown. The filters are expressed in a format that is compatible with `pyarrow.compute.Expression`
         :return: the PyArrow table
         """
+        if filters is not None:
+            filters = _filters_to_expression(filters)
         return self.to_pyarrow_dataset(
             partitions=partitions, filesystem=filesystem
-        ).to_table(columns=columns)
+        ).to_table(columns=columns, filter=filters)
 
     def to_pandas(
         self,
         partitions: Optional[List[Tuple[str, str, Any]]] = None,
         columns: Optional[List[str]] = None,
         filesystem: Optional[Union[str, pa_fs.FileSystem]] = None,
+        filters: Optional[
+            Union[List[Tuple[str, str, Any]], List[List[Tuple[str, str, Any]]]]
+        ] = None,
```

Comment:
Could we pull out the DNF filter type definition into a variable above? I feel like that would help with readability.

Reply:
> I feel like that would help with readability.

Yes, we can pull out it into variable. I will fix this.